src/components/PageComponents.js

```diff
@@ -60,15 +63,34 @@ class PageComponents extends Component {
     this.props.dispatch(uiBrowseUpdateList({ update: component, value: newComponent }))
   }
 
+  facetChange(value) {
+    const activeFacets = [];
+    value && value.forEach(facet => activeFacets.push(facet['value']))
+    this.setState({ ...this.state, activeFacets})
+    console.log('Current state in Page Component:', JSON.stringify(this.state))
+  }
+
+  renderActionButton() {
+    return (<Button className='pull-right' bsStyle='success' onClick={this.filterHandler}>Filter</Button>)
+  }
+
   noRowsRenderer() {
     return <div>Select components from the list above ...</div>
   }
 
   render() {
     const { components, filterOptions, definitions, token } = this.props
+    const { activeFacets } = this.state
     return (
       <Grid className='main-container'>
         <Row className='show-grid spacer'>
+          <Col md={4}>
+            <FacetSelect 
+              name="facets"
+              changeHandler={this.facetChange}
```

Comment:
typical name would be `onChange`. Not a rule, just seems to common "event style" naming